### PR TITLE
net: lwm2m: Allow to acknowledge request early from the callback

### PR DESCRIPTION
**Pull request by rlubos**
LwM2M engine by default sends piggybacked responses for requests after
all callbacks are executed. This approach however isn't good enough if
the application callback executes some lenghty operations (for instance
during FW update). Delaying the ACK may result in unnecessary
retransmissions.

This PR adds an API function which allows to send an early empty
ACK from the application callback. This prevents further retransmissions
from the server side. After all callbacks are executed, the LwM2M engine
will send the response as a separate CON message.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

File: include/net/lwm2m.h [hunk 1]
```diff
@@ -79,6 +79,7 @@ struct lwm2m_ctx {
 	struct coap_reply replies[CONFIG_LWM2M_ENGINE_MAX_REPLIES];
 	struct k_delayed_work retransmit_work;
 	struct sys_mutex send_lock;
+	void *processed_req;
```
Comment (jukkar) [inline include/net/lwm2m.h:82]: Can't this be `struct lwm2m_message`?

Comment (rlubos) [reply include/net/lwm2m.h:82]: `struct lwm2m_message` is declared in a private LwM2M header (https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/lwm2m/lwm2m_object.h#L411) and has dependencies to some other structures declared in there. Using it here would require to make it available in the public header, along with all these other structures it consists of and I'd rather avoid that.

Comment (jukkar) [reply include/net/lwm2m.h:82]: Hmm, ok. In that case could you add comment there why it is void and what is the correct type, now it is not obvious why we do not have proper type there.

Comment (rlubos) [reply include/net/lwm2m.h:82]: Ok, I can do that.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**